### PR TITLE
DatastreamToSpanner integration test - Interleaved table, FKs and Indexes

### PR DESCRIPTION
**Pull request by darshan-sj**
DatastreamToSpanner integration test - Interleaved table, FKs and Indexes

### CODE REVIEW

File: v2/datastream-to-spanner/src/test/resources/DataStreamToSpannerSimpleIT/mysql-schema.sql [hunk 1]
```diff
@@ -15,3 +15,28 @@ CREATE TABLE `Users` (
     `startDate` date,      -- To: startDate DATE
     PRIMARY KEY (`id`)
 );
+
```
Comment (shreyakhajanchi) [inline v2/datastream-to-spanner/src/test/resources/DataStreamToSpannerSimpleIT/mysql-schema.sql:18]: Why is there no index in source schema? I see it in session file

File: v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java [hunk 2]
```diff
@@ -189,6 +189,54 @@ public void migrationTestWithInsertsOnly() {
...
+                    SpannerRowsCheck.builder(spannerResourceManager, "Books")
+                        .setMinRows(3)
+                        .setMaxRows(3)
+                        .build(),
+                    SpannerRowsCheck.builder(spannerResourceManager, "Authors")
+                        .setMinRows(3)
+                        .setMaxRows(3)
+                        .build()))
+            .build();
+
```
Comment (shreyakhajanchi) [inline v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java:225]: We should also add unhappy test case of trying to insert record in child table before inserting in parent table

File: v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java [hunk 3]
```diff
@@ -189,6 +189,54 @@ public void migrationTestWithInsertsOnly() {
...
+    // Wait for conditions
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(12)), conditionCheck);
+
+    // Assert Conditions
+    assertThatResult(result).meetsConditions();
+
+    // Assert specific rows
+    assertAuthorsTable();
```
Comment (shreyakhajanchi) [inline v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java:235]: To validate that indexes are created and have data will it be better to query the indexes like below?
`SELECT AlbumId, AlbumTitle, MarketingBudget
    FROM Albums@{FORCE_INDEX=AlbumsByAlbumTitle}`

File: v2/datastream-to-spanner/src/test/resources/DataStreamToSpannerSimpleIT/mysql-schema.sql (hunk 1 above)
Comment (darshan-sj) [reply v2/datastream-to-spanner/src/test/resources/DataStreamToSpannerSimpleIT/mysql-schema.sql:18]: Done.

File: v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java (hunk 3 above)
Comment (darshan-sj) [reply v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java:235]: Done.

File: v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java (hunk 2 above)
Comment (shreyakhajanchi) [reply v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/DataStreamToSpannerSimpleIT.java:225]: In that case shouldn't the insertion of child records fail?

**Review by shreyakhajanchi — APPROVED** [review]
LGTM

**Pull request merged**